### DISCUSSION TITLE
negative reflectance values (from acolite processing)

### DISCUSSION THREAD

**Discussion by NOC-EO** [semiautomaticgit/SemiAutomaticClassificationPlugin#265] [General]
Hello Luca,

You have been generous with your time previously to answer my questions so thank you. I think SCP is great and I have been using it to classify seagrass in UK which is quite tricky due to sediment in the water column and other factors.

I have two questions:

(1) after using the acolite sea-glint correction I get negative values of reflectance and I think that this means I cannot use SCP - am I correct that any negative values have to be filtered out before using SCP

(2) I have never had any luck in reimporting a set of training polygons and then re-calculating the spectral signatures - so I have to keep digitising the same training polygons every time - I have tried loading an existing training dataset from teh SCP dock and also importing them via the Import signatures feature but I can t get anything to work - ideally I would simply import the polygons (not the signatures) and calculate the signatures for different images - maybe I am doing something wrong

However, overall your plugin is amazing and I am totally in favor of open-source solutions so thank you - Clive

**Comment by semiautomaticgit** [comment]
Hello @NOC-EO ,

thank you very much for your kind words and for using SCP.

About your questions:

> (1) after using the acolite sea-glint correction I get negative values of reflectance and I think that this means I cannot use SCP - am I correct that any negative values have to be filtered out before using SCP

You can still use SCP even with negative reflectance. However, you may want to clip the values between 0 and 1 using the tool [Band calc](https://semiautomaticclassificationmanual.readthedocs.io/en/latest/bandCalcTab.html) and the conditional operator `where`

> (2) I have never had any luck in reimporting a set of training polygons and then re-calculating the spectral signatures - so I have to keep digitising the same training polygons every time - I have tried loading an existing training dataset from teh SCP dock and also importing them via the Import signatures feature but I can t get anything to work - ideally I would simply import the polygons (not the signatures) and calculate the signatures for different images - maybe I am doing something wrong

Usually you should be able to import previously saved training polygons or vectors using this tool https://semiautomaticclassificationmanual.readthedocs.io/en/latest/importSignaturesTab.html#import-vector . Please check that projections of vector and raster files are matching.

**Comment by NOC-EO** [comment]
Hello - thanks for the reply - my main issue is that I if I create a training dataset for a particular image and classify an image I have then opened a new bandset and when I highlight the classes that I want o recalculate signatures fr and hit the correct button nothing happens.

I would like to be able to recalculate the signatures using exactly the same training polygons for several different images and despite trying everything I can think of so far I have not been able to do this.

many thanks
